Products/ZenWidgets/browser/Portlets.py

```diff
@@ -91,16 +91,16 @@ def getDevProdStateJSON(self, prodStates=['Maintenance']):
         devroot = self.context.dmd.Devices
         if isinstance(prodStates, basestring):
             prodStates = [prodStates]
+
         orderby, orderdir = 'id', 'asc'
         catalog = getattr(devroot, devroot.default_catalog)
-        queries = []
-        for state in prodStates:
-            queries.append(Eq('getProdState', state))
-        query = Or(*queries)
+        query = Indexed('id') # Get all the brains
         objects = catalog.evalAdvancedQuery(query, ((orderby, orderdir),))
+
         devs = (x.getObject() for x in objects)
         mydict = {'columns':['Device', 'Prod State'], 'data':[]}
         for dev in devs:
+            if not dev.getProductionStateString() in prodStates: continue
```

Comment:
This is too slow, use the global catalog so we don't have to wake up every device